src/widget/wtrackmenu.cpp

```diff
@@ -1781,31 +1792,53 @@ void WTrackMenu::slotRemoveFromDisk() {
         // Show purge summary message
         QMessageBox msgBoxPurgeTracks;
         msgBoxPurgeTracks.setIcon(QMessageBox::Information);
-        msgBoxPurgeTracks.setWindowTitle(tr("Track Files Deleted"));
-        msgBoxPurgeTracks.setText(
-                tr("%1 track files were deleted from disk and purged "
-                   "from the Mixxx database.")
-                        .arg(QString::number(tracksToPurge.length())) +
-                QString("<br><br>") +
-                tr("Note: if you are in Browse or Recording you need to "
-                   "click the current view again to see changes."));
+        QString msgTitle;
+        QString msgText;
+        if (m_pTrackModel) {
+            msgTitle = tr("Track Files Deleted");
+            msgText =
+                    tr("%1 track files were deleted from disk and purged "
+                       "from the Mixxx database.")
+                            .arg(QString::number(tracksToPurge.length())) +
+                    QString("<br><br>") +
+                    tr("Note: if you are in Browse or Recording you need to "
+                       "click the current view again to see changes.");
+        } else {
+            // Eject track from deck
+            ControlProxy* ejectTrackProxy = new ControlProxy(
+                    m_deckGroup, "eject", this);
+            ejectTrackProxy->set(1.0);
```

Comment:
I'm not sure this is the best way to do it, but the first and simplest that came to mind.

Reply:
Do we need to check if it's the deleted track that's loaded?
The progress dialog is modal so manual unloading is not possible in he GUI, but with keyboards and controllers.

Reply:
Deleting an MP3 file while the audio stream is decoded might crash Mixxx: https://bugs.launchpad.net/mixxx/+bug/1452005

This happens because libmad/SoundSourceMP3 uses memory mapped files. You are not able to detect how many decoders are active. Checking if a track is loaded on a deck is not sufficient.

Better drop libmad/SoundSourceMP3 and use something more robust.

Reply:
This might also happen with the existing delete & purge action, right? Like when the Analysis feature is busy.

How do you propose to proceed?